src/wattsi.pas

```diff
@@ -934,6 +946,18 @@    TCrossReferences = record
             Result := False;
          end
          else
+         if (Element.isIdentity(nsHTML, eA) and (Element.GetAttribute('class').AsString = 'sha-link') and (Variant = vSnap)) then
+         begin
+            Scratch := Default(Rope);
+            Scratch.Append(@SourceGitSHA[1], Length(SourceGitSHA));
+            Element.AppendChild(TText.CreateDestructively(Scratch));
+            Element.AppendChild(TText.Create(' commit'));
+            Scratch := Default(Rope);
+            Scratch.Append('https://github.com/whatwg/html/commit/');
+            Scratch.Append(@SourceGitSHA[1], Length(SourceGitSHA));
+            Element.SetAttributeDestructively('href', Scratch);
```

Comment:
This bit here doesn't work. End up with a bunch of wrong bytes in the final output. Elsewhere I could work around combining an AnsiString with some other type of string, but that doesn't work when setting attributes. I'd appreciate some help.

cc @Hixie @sideshowbarker